data/pools/transactionPool.go

```diff
@@ -894,7 +902,15 @@ func (pool *TransactionPool) assembleEmptyBlock(round basics.Round) (assembled *
 	next := bookkeeping.MakeBlock(prev)
 	blockEval, err := pool.ledger.StartEvaluator(next.BlockHeader, 0)
 	if err != nil {
-		err = fmt.Errorf("TransactionPool.assembleEmptyBlock: cannot start evaluator for %d: %v", round, err)
+		var nonSeqBlockEval ledgercore.ErrNonSequentialBlockEval
+		if errors.As(err, &nonSeqBlockEval) {
+			if nonSeqBlockEval.EvaluatorRound <= nonSeqBlockEval.LatestRound {
+				// in the case that the ledger have already moved beyond that round, just let the agreement know that
+				// we don't generate a block and it's perfectly fine.
+				return nil, ErrStaleBlockAssemblyRequest
```

Comment:
In one instance, the caller of this function `AssembleBlock()` will modify this error like this.
```
emptyBlockErr = fmt.Errorf("AssembleBlock: failed to construct empty block : %v", emptyBlockErr)
```

However, the caller of `AssembleBlock()` in `node/node.go` will check 
```
if err == pools.ErrStaleBlockAssemblyRequest {
```

We need to change `%v` to `%w` and change the equality to `errors.As()`.

Reply:
yes, I agree.

Reply:
right